addons/mail/static/src/models/messaging_menu/messaging_menu.js

```diff
@@ -34,7 +34,14 @@ function factory(dependencies) {
         /**
          * Toggle whether the messaging menu is open or not.
          */
-        toggleOpen() {
+        async toggleOpen() {
+            if (!this.isOpen) {
+                const inbox = this.env.messaging.inbox;
+                if (!inbox.mainCache.isLoaded && !inbox.mainCache.isLoading) {
```

Comment:
It doesn't change the issue that this is going to toggle based on the state in the future when the rpc is resolved and not on the state at the time of the click.

Reply:
I'm not 100% sure to understand what you mean. Is it exactly the same problem that you have with other toggles in general, or something else? Because the former I already argued that I feel code is much more maintainable the sooner a user interaction is dispatched in models, and toggle does this so it's the much better approach in my opinion.

Reply:
As discussed, at the bottom of the method there is the actual toggle, it shouldn't be done after "await" rpc.